src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Simplification/Simplifiers/CastSimplifier.cs

```diff
@@ -358,12 +358,13 @@ private static bool CastRemovalCouldCauseSignExtensionWarning(ExpressionSyntax c
             //
             // Then the original code has an implicit user defined conversion in it.  We can only remove this
             // if the new code would have the same conversion as well.
-            if (originalConversionOperation.Parent is IConversionOperation { IsImplicit: true, Conversion: { IsUserDefined: true } } originalParentImplicitConversion)
+            if (originalConversionOperation.Parent is IConversionOperation { Conversion.IsUserDefined: true } originalParentConversion &&
+                originalParentConversion.GetConversion().IsImplicit)
```

Comment:
sigh.  boneheaded mistake on my part.  IConversionOperation.IsImplicit doesn't mean it's an implciit conversion.  THis is .IsImplicit inherited from IOperation (where it means the operation was not written in source).